distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Extensions.enso

```diff
@@ -840,58 +841,119 @@ Text.drop_last count =
+         "Hello World!".take (Range -3 Nothing) == "ld!"
+         "Hello World!".take (Range 5 Nothing) == " World!"
+         "Hello World!".take (Range 5 12) == " World!"
+         "Hello World!".take (Range 12 12) == ""
+Text.take : (Text_Sub_Range | Range) -> Text ! Index_Out_Of_Bounds_Error
+Text.take range =
+    char_range = case range of
+        Range _ _ -> here.range_to_char_indices this range
+        _ -> range.to_char_range this
+    if char_range.is_error then char_range else
+        Text_Utils.substring this char_range.start char_range.end
+
+## Creates a new Text by removing the specified range of the input.
+
+   This can select a section of text from the beginning, end, or middle of the
+   input using various criteria defined by the range parameter.
 
-   ! What is a Character?
-     A character is defined as an Extended Grapheme Cluster, see Unicode
-     Standard Annex 29. This is the smallest unit that still has semantic
-     meaning in most text-processing applications.
-
-   > Example
-     Make a new text from the last two characters of "boo".
-
-         "boo".take_last 2
-Text.take_last : Integer -> Text
-Text.take_last count =
-    iterator = BreakIterator.getCharacterInstance
-    iterator.setText this
-    iterator.last
-    boundary = iterator.next -count
-    if boundary == -1 then this else Text_Utils.drop_first this boundary
+   Arguments:
+   - range: The section of the this text to return.
+     If a `Text_Sub_Range`, then the selection is interpreted following the rules of that type.
+     If a `Range`, the selection is specified by two indices, from and to.
+
+   Returns:
+   The part of the input as specified by the range parameter.
+
+   > Examples
+     Various different ways to take part of "Hello World!"
+
+         "Hello World!".drop First.new == "ello World!"
+         "Hello World!".drop (First 5) == " World!"
+         "Hello World!".drop (First 0) == "Hello World!"
+         "Hello World!".drop Last.new == "Hello World"
+         "Hello World!".drop (Last 6) == "Hello "
+         "Hello World!".drop (Before " ") == " World!"
+         "Hello World!".drop (Before_Last "o") == "orld!"
+         "Hello World!".drop (After " ") == "Hello "
+         "Hello World!".drop (After_Last "o") == "Hello Wo"
+         "Hello World!".drop (While c->c!=" ") == " World!"
+         "Hello World!".drop (Range 3 5) == "Hel World!"
+         "Hello World!".drop (Range -3 -1) == "Hello Wor!"
+         "Hello World!".drop (Range -3 Nothing) == "Hello Wor"
+         "Hello World!".drop (Range 5 Nothing) == "Hello"
+         "Hello World!".drop (Range 5 12) == "Hello"
+         "Hello World!".drop (Range 12 12) == "Hello World!"
+Text.drop : (Text_Sub_Range | Range) -> Text ! Index_Out_Of_Bounds_Error
+Text.drop range =
+    char_range = case range of
+        Range _ _ -> here.range_to_char_indices this range
+        _ -> range.to_char_range this
+    if char_range.is_error then char_range else
```

Comment:
I don't understand this check. If `char_range` is an error, you just pass it through. But if it was an error and then the first `if` does a check on it - it would fail and pass the dataflow error anyway. So why do this check here? Seems unnecessary unless I'm missing something.

Also, `take` does not have an analogous one, so if anything - it should be consistent.

Reply:
`drop` has to invert the selection to make the output. 
As you say this line was unneeded so have removed.